csrc/index_compute.cpp

```diff
@@ -1601,12 +1603,15 @@ std::vector<Val*> Index::getGlobalProducerStridedIndices(
     for (const auto i : c10::irange(alloc_dom.size())) {
       if (alloc_dom[i]->isReduction()) {
```

Comment:
a one-line patch should be sufficient here, right?
```suggestion
      if (alloc_dom[i]->isReduction() || alloc_dom[i]->isDeviceDim()) {
```

Reply:
Not quite. The `isReduction` branch doesn't increment `stride_i` since reduction iter domains are not included in allocation domain (someone please correct me if I'm wrong), but the `isDeviceDim` branch increments `stride_i` since sharded iter domains are in the allocation domain.